File: src/code_without_input/128.最长连续序列.cpp
/*
 * @lc app=leetcode.cn id=128 lang=cpp
 *
 * [128] 最长连续序列
 */

// @lc code=start
#include <vector>
#include <algorithm>
#include <unordered_set>

using namespace std;
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> numSet(nums.begin(), nums.end());
        int longestStreak = 0;

        for (int num : numSet) {
            if (numSet.find(num - 1) == numSet.end()) {
                int currentNum = num;
                int currentStreak = 1;

                while (numSet.find(currentNum + 1) != numSet.end()) {
                    currentNum++;
                    currentStreak++;
                }

                longestStreak = max(longestStreak, currentStreak);
            }
        }

        return longestStreak;
    }
};
// @lc code=end

File: src/code_without_input/128.最长连续序列.java
/*
 * @lc app=leetcode.cn id=128 lang=java
 *
 * [128] 最长连续序列
 */

// @lc code=start

import java.util.HashSet;
import java.util.Set;

class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> tmp = new HashSet<>();
        for(int num : nums){
            tmp.add(num);
        }
        int max = 0;
        for(int num : tmp){
            if(tmp.contains(num - 1)){
                continue;
            }else{
                
                int next = num + 1;
                while(tmp.contains(next)){
                    
                    next++;
                }
                max = Math.max(max, next - num);
            }
        }
        return max;
    }
}
// @lc code=end

